workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/compute.py

```diff
@@ -45,148 +40,44 @@
     RMSE_VARS,
     PRESSURE_INTERPOLATED_VARS,
 )
+from .registry import Registry
 
 import logging
 
 logger = logging.getLogger("SaveDiags")
-
-_DIAG_FNS = defaultdict(list)
-
-DiagDict = Mapping[str, xr.DataArray]
-
-
-def _start_logger_if_necessary():
-    # workaround for joblib.Parallel logging from
-    # https://github.com/joblib/joblib/issues/1017
-    logger = logging.getLogger("SaveDiags")
-    if len(logger.handlers) == 0:
-        logger.setLevel(logging.INFO)
-        sh = logging.StreamHandler()
-        sh.setFormatter(logging.Formatter("%(asctime)s %(levelname)-8s %(message)s"))
-        fh = logging.FileHandler("out.log", mode="w")
-        fh.setFormatter(logging.Formatter("%(asctime)s %(levelname)-8s %(message)s"))
-        logger.addHandler(sh)
-        logger.addHandler(fh)
-    return logger
+xr.set_options(keep_attrs=True)
 
 
-def _prepare_diag_dict(
-    suffix: str, new_diags: xr.Dataset, src_ds: xr.Dataset
-) -> DiagDict:
+def _prepare_diag_dict(suffix: str, ds: xr.Dataset) -> Mapping[str, xr.DataArray]:
     """
-    Take a diagnostic dict add a suffix to all variable names and transfer attributes
-    from a source dataset.  This is useful when the calculated diagnostics are a 1-to-1
-    mapping from the source.
+    Take a diagnostic dataset and add a suffix to all variable names and return as dict.
     """
 
     diags = {}
-    for variable in new_diags:
+    for variable in ds:
         lower = variable.lower()
-        da = new_diags[variable]
-        if variable in src_ds:
-            logger.debug(
-                "Transferring available diagnostic attributes from source for "
-                f"{variable}."
-            )
-            src_attrs = src_ds[variable].attrs
-            da = da.assign_attrs(src_attrs)
-        else:
-            logger.debug(
-                f"Not transferring attributes from source dataset for {variable}. This "
-                "may cause issues with automated report generation."
-            )
-
+        da = ds[variable]
```

Comment:
I made it redundant by setting `xr.set_options(keep_attrs=True)` in this module and in `transform.py`. The histogram one needs to be done manually, since the `vcm` function uses numpy.